uber/models/department.py

```diff
@@ -0,0 +1,271 @@
+from datetime import timedelta
+
+from sideboard.lib import cached_property
+from sideboard.lib.sa import CoerceUTF8 as UnicodeText, \
+    UTCDateTime, UUID
+from sqlalchemy import select
+from sqlalchemy.orm import column_property
+from sqlalchemy.schema import ForeignKey, Table, UniqueConstraint
+from sqlalchemy.types import Boolean, Float, Integer
+
+from uber.config import c
+from uber.models import MagModel
+from uber.models.attendee import Attendee
+from uber.models.types import default_relationship as relationship, \
+    Choice, DefaultColumn as Column
+
+
+__all__ = [
+    'department_membership_job_role', 'department_membership_request',
+    'job_required_job_role', 'Department', 'DepartmentMembership', 'Job',
+    'JobRole', 'Shift']
+
+
+department_membership_job_role = Table(
+    'department_membership_job_role',
+    MagModel.metadata,
+    Column('department_membership_id', UUID,
+           ForeignKey('department_membership.id')),
+    Column('job_role_id', UUID, ForeignKey('job_role.id')))
+
+
+department_membership_request = Table(
+    'department_membership_request',
+    MagModel.metadata,
+    Column('attendee_id', UUID, ForeignKey('attendee.id')),
+    Column('department_id', UUID, ForeignKey('department.id'), nullable=True),
+    UniqueConstraint('attendee_id', 'department_id'))
+
+
+job_required_job_role = Table(
+    'job_required_job_role',
+    MagModel.metadata,
+    Column('job_id', UUID, ForeignKey('job.id')),
+    Column('job_role_id', UUID, ForeignKey('job_role.id')))
+
+
+class DepartmentMembership(MagModel):
+
+    DEPARTMENT_HEAD = 100
+    TRUSTED = 200
+    REGULAR = 300
+    BANNED = 500
+
+    STATUS_OPTS = [
+        (DEPARTMENT_HEAD, 'Department Head'),
+        (TRUSTED, 'Trusted'),
+        (REGULAR, 'Regular'),
+        (BANNED, 'Banned')]
+    STATUSES = dict(STATUS_OPTS)
```

Comment:
This is missing the third, rarely-used config var we create in config.py, which is `_VARS`. That's not necessarily a problem, just wanted to make sure it wasn't an oversight.